cpp/examples/nested_types/deduplication.cpp

```diff
@@ -0,0 +1,173 @@
+  auto val  = tbl.column(0);
+  cudf::groupby::groupby grpby_obj(keys);
+  std::vector<cudf::groupby::aggregation_request> requests;
+  requests.emplace_back(cudf::groupby::aggregation_request());
+  auto agg = cudf::make_count_aggregation<cudf::groupby_aggregation>();
+  requests[0].aggregations.push_back(std::move(agg));
+  requests[0].values = val;
+  auto agg_results   = grpby_obj.aggregate(requests);
+  auto result_key    = std::move(agg_results.first);
+  auto result_val    = std::move(agg_results.second[0].results[0]);
+
+  auto left_cols = result_key->release();
+  left_cols.push_back(std::move(result_val));
+  // std::vector<cudf::column_view> columns{result_key->get_column(0), *result_val};
+  // auto agg_v = cudf::table_view(columns);
+
+  // Join on keys to get
+  return std::make_unique<cudf::table>(std::move(left_cols));
+}
+
+std::unique_ptr<cudf::table> join_count(cudf::table_view left, cudf::table_view right)
+{
+  auto [left_indices, right_indices] =
+    cudf::inner_join(cudf::table_view{{left.column(0)}}, cudf::table_view{{right.column(0)}});
+  auto new_left  = cudf::gather(left, cudf::device_span<int const>{*left_indices});
+  auto new_right = cudf::gather(right, cudf::device_span<int const>{*right_indices});
+
+  auto left_cols  = new_left->release();
+  auto right_cols = new_right->release();
+  left_cols.push_back(std::move(right_cols[1]));
+
+  return std::make_unique<cudf::table>(std::move(left_cols));
+}
+
+std::unique_ptr<cudf::table> sort_keys(cudf::table_view tbl)
+{
+  auto sort_order = cudf::sorted_order(cudf::table_view{{tbl.column(0)}});
+  return cudf::gather(tbl, *sort_order);
+}
+
+/**
+ * @brief Main for nested_types examples
+ *
+ * Command line parameters:
+ * 1. JSON input file name/path (default: "example.json")
+ * 2. JSON output file name/path (default: "output.json")
+ * 3. Memory resource (optional): "pool" or "cuda" (default: "pool")
+ *
+ * The stdout includes the number of rows in the input and the output size in bytes.
+ */
+int main(int argc, char const** argv)
+{
+  std::string input_filepath;
+  std::string output_filepath;
+  std::string mr_name;
+  if (argc < 4) {
+    input_filepath  = "example.json";
+    output_filepath = "output.json";
+    mr_name         = "pool";
+  } else if (argc == 4) {
+    input_filepath  = argv[1];
+    output_filepath = argv[2];
+    mr_name         = argv[3];
+  } else {
+    std::cout << "Either provide all command-line arguments, or none to use defaults" << std::endl;
+    return 1;
+  }
+
+  auto resource = create_memory_resource(mr_name);
+  rmm::mr::set_current_device_resource(resource.get());
+
+  // read input file
+  auto [tbl, metadata] = read_json(input_filepath);
+
+  auto st = std::chrono::steady_clock::now();
+
+  auto count                               = count_aggregate(tbl->view());
+  std::chrono::duration<double> count_time = std::chrono::steady_clock::now() - st;
+  std::cout << "Wall time: " << count_time.count() << " seconds\n";
```

Comment:
The example output would look like this:
```bash
Wall time: 1.0 seconds
Wall time: 2.0 seconds
Wall time: 3.0 seconds
Wall time: 4.0 seconds
```
Could we print out more specific timing info? something like (just the rough idea, you see what I mean):
```bash
Count aggregation time: 1.0 seconds
Join time: 2.0 seconds
Sort time: ...
```

Reply:
One comment from the console parsing side, using the pattern: 
`{func_name} time: X seconds` is easier to parse on a `.split(' ')`